enhancements/kubelet/virtualization-higher-workload-density.md

```diff
@@ -0,0 +1,551 @@
+---
+title: virtualization-higher-workload-density
+authors:
+  - "@enp0s3"
+  - "@iholder101"
+  - "@fabiand"
+reviewers:
+  - "@rphilips"
+  - "@sjenning"
+approvers:
+  - "@mrunalp"
+api-approvers:
+  - "@mrunalp"
+creation-date: "2024-05-05"
+last-updated: "2024-06-12"
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - "https://issues.redhat.com/browse/CNV-28178"
+see-also:
+  - "https://github.com/kubernetes/enhancements/issues/2400"
+status: implementable
+---
+
+# Virtualization higher workload density
+
+## Summary
+
+Fit more workloads onto a given node - achieve a higher workload
+density - by overcommitting it's memory resources. Due to timeline
+needs a two-phased approach is considered.
+
+## Motivation
+
+Today, OpenShift Virtualization is reserving memory (`requests.memory`)
+according to the needs of the virtual machine and it's infrastructure
+(the VM related pod). However, usually an application within the virtual
+machine does not utilize _all_ the memory _all_ the time. Instead,
+only _sometimes_ there are memory spikes within the virtual machine.
+And usually this is also true for the infrastucture part (the pod) of a VM:
+Not all the memory is used all the time. Because of this assumption, in
+the following we are not differentiating between the guest of a VM and the
+infrastructure of a VM, instead we are just speaking colectively of a VM.
+
+Now - Extrapolating this behavior from one to all virtual machines on a
+given node leads to the observation that _on average_ there is no memory
+ressure and often a rather low memory utilization - despite the fact that
+much memory has been reserved.
+Reserved but underutilized hardware resources - like memory in this case -
+are a cost factor to cluster owners.
+
+This proposal is about increasing the virtual machine density and thus
+memory utilization per node, in order to reduce the cost per virtual machine.
+
+> [!NOTE]
+> CPU resources are already over-committed in OpenShift today, thus the
+> limiting factor - usually - are memory resources.
+
+### User Stories
+
+* As an administrator, I want to be able to enable higher workload
+  density for OpenShift Virtualization clusters so that I can increase
+  the memory utilization.
+
+### Goals
+
+#### Timeline
+
+* GA higher workload density in OpenShift Virtualization in 2024
```

Comment:
is this timeline still accurate?

Reply:
@haircommander Hi, I've checked the Jira planning, we are on track, so yes this is indeed accurate.

Reply:
@haircommander The timeline bullet `GA higher workload density in OpenShift Virtualization in 2024` relates to the phase 1 only. Maybe we should add it in brackets

Reply:
Hey @haircommander!
@enp0s3 and I reworked the PR. Can you please have another look?